### ISSUE TITLE
Unable to log in through app, but login through webinterface works fine

### ISSUE THREAD

**Issue by vonshednob** [gotify/android#201]
It’s probably something very obvious that I’m doing wrong here, but I can’t quite find it.

So I set up my own server and and reach it fine through the webinterface, even from my android device.

However, when I try to use the app (version 2.2.0, installed from f-droid), I get the "There is no user with this username and password" error message. In the server logs, I can also see Error#01 for the access to `/current/user`.

Sorry if this is too obvious and could be found in the documentation. I didn’t find it.

**Comment by jmattheis** [comment]
It could be related to https://github.com/gotify/android/issues/161, can you check if the issue still occurs with a password that only contains letters.

**Comment by vonshednob** [comment]
The password was `test`, so it’s not immediately connected to #161, I think.

**Comment by jmattheis** [comment]
Have you used some exotic username, that contained special characters?

**Comment by vonshednob** [comment]
I was thinking the same, but it’s ANSI only, too. In line with some testing attempts you can probably guess the (hence deleted) user’s name: `test`.

**Comment by jmattheis** [comment]
Hmpf, this should work. How have you started gotify, is there any reverse proxy in front of gotify?

**Comment by vonshednob** [comment]
Alright, some more on the setup: gotify is running on a machine in the internal network and an nginx proxy is forwarding and de-https’ing the requests for gotify.

I was wondering whether the sub-path on the nginx proxy might be the problem. It’s some `/obscure/.sub-path/thingy` (as close to reality as can be).
However, I see the requests from the android client arriving at the server (tail’ing the log).

**Comment by vonshednob** [comment]
So, looks like there was another step of redirection in between that got lost in translation.

The entry point is `example.org/obscure/.sub-path/thingy`, but it will `301 permanent` to `proxy.example.org/obscure/.sub-path/thingy`. The latter is the nginx proxy as described above.

Now, when I use the `example.org` entry point in the app, it fails to log in (however, it correctly detects that there’s a gotify service running there). But if I use `proxy.example.org` as the entry point, it works!

That’s probably a bug?

**Comment by jmattheis** [comment]
I'd say this is undefined behavior. It could probably be fixed, but I'd guess no one actually has a use-case for this, and can just use a URL that doesn't redirect.

**Comment by vonshednob** [comment]
Fair enough. My irritation stems from the fact that it’s working to some extend (requests seem to go through, as I can see some requests arrive from the android client on the server) and that the website works.

However, since there is a work-around, I don’t see any urgency :)

Thanks a lot for the software!

**Comment by novaliswolf** [comment]
Hi, just wanted to mention that I've ran into this issue as well. This is what I have set up:

1. Gotify running in docker
2. Traefik acting as a revers proxy redirecting HTTP traffic to HTTPS for gotify and assigning a letsencrypt certificate (for internal domain)
3. Exposed Gotify using a cloudflare tunnel which points to traefik. 

This all works internally/externally on a web browser, but for some reason on the android app it gives me the same error message as @vonshednob. 

Unfortunately even after a lot of testing I couldn't get it to work, so I've had to map and expose the port outside of traefik in the dockerd compose file specifically for Gotify and changed the clouldflare tunnel so that it goes to http://$dockerHostIp:$mappedPort directly and that works.

I don't know enough to say what the problem is but it does feel like something to do with the request going through a reverse proxy that's making the android app unhappy (as I said via mobile browser I can login using the original method)

Thanks and happy new year!

**Comment by jmattheis** [comment]
@novaliswolf Have you configured teh https address in the android app? The android app should work fine with a reverse proxy as long as there are no redirects. Would you mind sharing your gotify url (you can do so privately via the email on my profile).
